python/imzml_reader.py

```diff
@@ -183,10 +193,22 @@ def write_metadata_json(json_file):
         required=True,
         help="Write the metadata about the IMS zarr store on S3.",
     )
+    parser.add_argument(
+        "--zarr_store_url",
+        required=True,
+        help="Destination for zarr output in cloud.",
+    )
     args = parser.parse_args()
 
     dataset = IMSDataset(
         args.imzml_file, args.ibd_file, micro_res=0.5, ims_res=10
     )
-    dataset.write_zarr(args.ims_zarr, dtype="i4", compressor=Zlib(level=1))
-    write_metadata_json(args.ims_metadata)
+    dataset.write_zarr(args.ims_zarr, dtype="uint32", compressor=Zlib(level=1))
```

Comment:
Is there any way to infer `dtype` from the data?  If so, I think it would be good so that this could be a bit more robust.